File: src/commands/nocrouch.ts
import { SlashCommandBuilder, ChatInputCommandInteraction } from "discord.js";
import { reply } from "../lib/functions/discord";

export default {
	data: new SlashCommandBuilder()
		.setName("nocrouch")
		.setDescription("Approximate potential distance of a nocrouch jump.")
		.addNumberOption((o) =>
			o
				.setName("distance")
				.setDescription("Specify the distance of your nocrouch jump.")
				.setRequired(true)
		)
		.addNumberOption((o) =>
			o
				.setName("max")
				.setDescription("Specify the max speed of your nocrouch jump.")
				.setRequired(true)
		),
	execute(interaction: ChatInputCommandInteraction) {
		const distance = interaction.options.getNumber("distance")!;
		const max = interaction.options.getNumber("max")!;
		const approx = distance + (max / 128) * 4;

		return reply(interaction, {
			content: `Approximated distance: \`${approx.toFixed(4)}\``
		});
	}
};

File: src/commands/nocrouch.rs
use {
	crate::{prelude::InteractionResult, events::interactions::InteractionState},
	serenity::{builder::CreateApplicationCommand, model::prelude::command::CommandOptionType},
};

pub(crate) fn register(cmd: &mut CreateApplicationCommand) -> &mut CreateApplicationCommand {
	return cmd
		.name("nocrouch")
		.description("Calculates an approximation for the potential distance of an uncrouched LJ.")
		.create_option(|opt| {
			opt.kind(CommandOptionType::Number)
				.name("distance")
				.description("The distance of your jump.")
				.required(true)
		})
		.create_option(|opt| {
			opt.kind(CommandOptionType::Number)
				.name("max")
				.description("The max speed of your jump.")
				.required(true)
		});
}

pub(crate) async fn execute(state: &InteractionState<'_>) -> InteractionResult {
	let distance = state.get::<f64>("distance").expect("This option is marked as `required`.");

	let max = state.get::<f64>("max").expect("This option is marked as `required`.");

	let result = distance + (max / 128.) * 4.;

	Ok(format!("Approximated distance: `{0:.4}`", result).into())
}
